src/user/proposals/UserProposalDashboard/DashboardContent/DashboardContent.js

```diff
@@ -178,90 +201,121 @@ class DashboardContent extends Component {
 
         <div className="proposal-container">
           <div className="proposals">
-            {this.state.displayItems.map((proposalItem, index) => {
-              return (
-                <div className="single-proposal" key={index}>
-                  <div className="user-info">
-                    <div className="image">
-                      <Image src={userIcon2} alt="icon" rounded />
-                    </div>
-                    <div className="img-desc">
-                      <h2>{proposalItem.title}</h2>
-                      <p className="proposal-date">{proposalItem.createdAt}</p>
-                    </div>
-                    {proposalItem.proposalStatus === "DRAFT" ? (
-                      <Button
-                        variant="primary"
-                        className="status-btn"
-                        size="sm"
-                      >
-                        <span className="btn-content">
-                          {proposalItem.proposalStatus}
-                        </span>
-                      </Button>
-                    ) : (
-                      <Button
-                        variant="primary"
-                        className="status-btn-submitted"
-                        size="sm"
-                      >
-                        <span className="btn-content">
-                          {proposalItem.proposalStatus}
-                        </span>
-                      </Button>
-                    )}
-                  </div>
-                  <div className="proposal-content">
-                    <div className="proposal-details">
-                      {proposalItem.proposalDescription}
+            {displayItems
+              .slice(currentPage * pageSize, (currentPage + 1) * pageSize)
```

Comment:
@AuraOfDivinity You have implemented pagination in backend itself, then why you are slicing the objects here??

Reply:
@Rupeshiya Fetch all proposals routes is not paginated in the backend bro.